### PR TITLE
Appendum #3574 - list of values

### PR DESCRIPTION
**Pull request by TracerDS**
The point of this PR is to create a type for ArgumentParser that will accept a tuple/CLuaMultiReturn:
```cpp

bool CLuaSomeDefs::SomeFunc(int a, CLuaMultiReturn<bool, bool, bool> args) noexcept
{
    // Guarantee that `args` will store 3 bools
    return CStaticFunctionDefinitions::DoSomethingWith(a, args.Get(0), args.Get(1), args.Get(2));
}
```

### CODE REVIEW

File: Server/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp [hunk 1]
```diff
@@ -16,9 +16,19 @@
...
+    return std::get<0>(ret.values) + std::get<1>(ret.values) + std::get<2>(ret.values);
+}
+#endif
+
 void CLuaElementDefs::LoadFunctions()
 {
     constexpr static const std::pair<const char*, lua_CFunction> functions[]{
+#ifdef ARGUMENT_PARSER_TUPLE_DEBUG
+        {"testFunc", ArgumentParser<a>},
+#endif
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp:31]: For debugging purposes. Will be removed when the PR is ready

File: Server/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp [hunk 2]
```diff
@@ -16,9 +16,19 @@
 #include "CDummy.h"
 #include "Utils.h"
 
+#ifdef ARGUMENT_PARSER_TUPLE_DEBUG
+float a(CLuaMultiReturn<float, float, float> ret)
+{
+    return std::get<0>(ret.values) + std::get<1>(ret.values) + std::get<2>(ret.values);
+}
+#endif
+
```
Comment (tederis) [inline Server/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp:25]: I don't think that a repository is a good place to keep this debug stuff.

Comment (TracerDS) [reply Server/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp:25]: > For debugging purposes. Will be removed when the PR is ready

Right now it is only to check if `CLuaMultiReturn` works correctly in ArgumentParser

**Comment by tederis** [comment]
Is this still a draft?

**Comment by TracerDS** [comment]
> Is this still a draft?

Yeah, I got stuck with this

**Comment by TracerDS** [comment]
Im not interested in advancing this PR further anymore.
If someone wants to refresh it, feel free to do so.

**Pull request closed without merging**